omg/entities/player.py

```diff
@@ -85,13 +106,28 @@ def update(self, mouse_x, mouse_y, delta_time):
         )
         self._regenerate_mana(delta_time)
 
-    def shoot(self):
-        """Shoots a projectile and informs the observes with an ProjectileShotEvent."""
-        if self.skills.current_size == 0 or self.current_mana < 20:
+        # Update combined elements
+        if len(self.to_be_combined_element_cash) == 2:
+            # Get the name of the skill after combining elements
+            new_skill = "".join(self.to_be_combined_element_cash[::])
+
+            # Remove "ElementFactory" from the element name
+            new_skill = new_skill.replace("ElementFactory", "")
+
+            # Update crafted skill slots
+            self._update_crafted_skill_slots(new_skill)
+
+            # Empty the element cash
+            self.to_be_combined_element_cash = []
+
+    def shoot(self, skill_name: str):
+        """Shoot a projectile and inform the observers."""
+        mana_cost = COMBINED_ELEMENT_DICTIONARY[skill_name].mana_cost
```

Comment:
since you did not type-hinted COMBINED_ELEMENT_DICTIONARY, my text editor cannot the return type of `COMBINED_ELEMENT_DICTIONARY[skill_name]`, then it makes it harder to read and analyze the code. For example `COMBINED_ELEMENT_DICTIONARY[skill_name].mana_cost` is not highlighted by the editor currently.

It is very clear that what `COMBINED_ELEMENT_DICTIONARY` would cover, please add type-hinting. ( but as I mentioned in the comments above, I am not sure if using COMBINED_ELEMENT_DICTIONARY in this way is the right way to go)